### PR TITLE
Solver Analytics CI

### PR DESCRIPTION
**Pull request by msaudulhassan**
SolverAnalytics (https://github.com/ROCmSoftwarePlatform/SolverAnalytics) depends on Tuna, and changes to Tuna might break SolverAnalytics. This PR adds a stage to Tuna's CI that'd run SolverAnalytics' tests to make sure any future Tuna PRs don't break SolverAnalytics.

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -25,6 +25,14 @@ pipeline {
...
     } 
     stages {
+        stage("solver analytics test") {
+        agent{  label utils.rocmnode("tunatest") }
+        steps {
+          script {
+            utils.solverAnalyticsTest()
+            }
+            }
+        }
```
Comment (iq136boy) [inline Jenkinsfile]: Maybe add "solver analytics test" after the 'code format' and 'pylint' stage? Normally, we would like to check the code format and style first, then do the other checks in the CI.

**Review by iq136boy — DISMISSED** [review]

File: Jenkinsfile (hunk 1 above)
Comment (msaudulhassan) [reply Jenkinsfile]: I'm trying to test out the solver analytics tests, that's why I moved it to the first stage. Once it's working, I'll move it to the end

File: Dockerfile [hunk 2]
```diff
@@ -176,3 +175,7 @@ ADD requirements.txt /tuna/
 WORKDIR /tuna
 
 RUN python3 setup.py install
+
+# Install SolverAnalytics (github.com/ROCmSoftwarePlatform/SolverAnalytics#installation)
+ARG SOLVER_ANALYTICS_DIR=/tuna/SolverAnalytics/
```
Comment (alexandraBara) [inline Dockerfile]: i think you might have to create the dir /tuna/SolverAnalytics and then try and clone in there. What is the error you are getting?  like:
RUN mkdir /tuna/SolverAnalytics

File: vars/utils.groovy [hunk 3]
```diff
@@ -316,6 +316,31 @@ def loadJobTest() {
...
+        checkout scm
+        env.TUNA_DB_HOSTNAME = "${db_host}"
+        env.TUNA_DB_NAME = "${db_name}"
+        env.TUNA_DB_USER_NAME = "${db_user}"
+        env.TUNA_DB_PASSWORD = "${db_password}"
+        env.gateway_ip = "${gateway_ip}"
+        env.gateway_port = "${gateway_port}"
+        env.gateway_user = "${gateway_user}"
+        env.PYTHONPATH = "${env.WORKSPACE}/tuna:${env.PYTHONPATH}"
+        env.PATH = "${env.WORKSPACE}/tuna:${env.PATH}"
```
Comment (alexandraBara) [inline vars/utils.groovy:332]: you probably want the PYTONPATH and the PATH to point to /tuna/SolverAnalytics right?

**Review by alexandraBara — APPROVED** [review]

**Review by JehandadKhan — APPROVED** [review]

**Pull request merged**